internal/scheduler/scheduler.go

```diff
@@ -140,17 +138,33 @@ func (s *Scheduler) UpdateJobNextRun(ctx context.Context, jobId JobId, nextRunIn
 }
 
 // Start begins the scheduling loop that will query the repository for jobs to run and
-// run them in a goroutine.
-func (s *Scheduler) Start(ctx context.Context) {
+// run them in a goroutine, the scheduler will stop all running jobs and stop requesting
+// new jobs once the ctx past in is cancelled.
+// The scheduler cannot be started again once the ctx is cancelled, a new scheduler will
+// need to be instantiated in order to begin scheduling again.
+func (s *Scheduler) Start(ctx context.Context) error {
+	const op = "scheduler.(Scheduler).Start"
 	if !s.started.CAS(s.started.Load(), true) {
 		s.logger.Debug("scheduler already started, skipping")
-		return
+		return nil
+	}
+
```

Comment:
Should a check be done to make sure `ctx` has not been canceled/closed before work starts? For example:

```go
if err := ctx.Err(); err != nil {
  return err
}
```